### PR TITLE
[WIP] Using react routers

### PR DESCRIPTION
**Pull request by pierr**
- [x] Migrate all routers to use React Router

### CODE REVIEW

File: app/initializer/scripts/global-link-initializer.js [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-    });
-}
+// // TODO rewrite : remove jquery
+// import $ from 'jquery';
+// import history from 'focus-core/history';
+//
+// export default () => {
+//     console.info('|--- GLOBAL LINKS');
+//
+//     $(document).on('click', 'a:not([data-bypass])', function touchHandler(evt) {
```
Comment (pierr) [inline app/initializer/scripts/global-link-initializer.js]: don't comment this remove the file it has no longer a meaning

File: app/initializer/scripts/layout-initializer.js [hunk 2]
```diff
@@ -4,11 +4,14 @@ import Layout from 'focus-components/components/layout';
...
 import DevTools from '../../components/dev-tools';
+import {Router, browserHistory} from 'react-router'
+import routes from '../../router/routes';
+
 export default () => {
     console.info('|--- LAYOUT');
 
     const CustomLayout = (props) => {
-      return <div><Layout MenuLeft={DemoMenuLeft} Footer={DemoFooter} /> <DevTools/></div>
+      return <div> <Layout MenuLeft={DemoMenuLeft} Footer={DemoFooter}></Layout> <DevTools/></div>
```
Comment (pierr) [inline app/initializer/scripts/layout-initializer.js]: It should be useless or there is a duplication with the first layout

File: app/views/home/index.jsx [hunk 3]
```diff
@@ -17,7 +18,8 @@ export default React.createClass({
     mixins: [cartridgeBehaviour],
     /** @inheritDoc */
     _navigateAdvancedSearch() {
-        history.navigate('#search/advanced', true);
```
Comment (pierr) [inline app/views/home/index.jsx:20]: it shoule be a props

File: app/views/menu/menu-left.jsx [hunk 4]
```diff
@@ -25,12 +29,16 @@ export default React.createClass({
 
     _onHomeClick() {
         this._onMenuItemClick();
-        history.navigate('#', true);
+        browserHistory.push('/');
```
Comment (pierr) [inline app/views/menu/menu-left.jsx]: It should be a props on the focus components menu

File: app/initializer/scripts/layout-initializer.js (hunk 2 above)
Comment (GuenoleK) [inline app/initializer/scripts/layout-initializer.js]: Yes it's now useless

**Pull request merged**